### PR TITLE
[FIX] trigger info not updating when clicking on another trigger

### PR DESCRIPTION
**Pull request by FabioCavaleti**
## Proposed changes (including videos or screenshots)

- Fix the bug in omnichannel triggers that the trigger info was not updating when switching the one that the info is displayed
- Refactor using react hook forms
- Typescript conversion
- Replace deprecated useEndpointData
- Added test to verify if the bug happens again

## Further comments

OC-591

### CODE REVIEW

File: apps/meteor/client/views/omnichannel/triggers/TriggersFormWithData.tsx [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+	if (isError || !data?.trigger) {
+		return <Callout>{t('Error')}: error</Callout>;
+	}
+
+	return (
+		<>
+			<TriggersForm initialValues={getInitialValues(data.trigger)} onSave={onSave} />
+		</>
+	);
+};
```
Comment (filipemarins) [inline apps/meteor/client/views/omnichannel/triggers/TriggersFormWithData.tsx:57]:
```suggestion
	return (<TriggersForm initialValues={getInitialValues(data.trigger)} onSave={onSave} />);
};
```
Is the fragment still required?

**Review by filipemarins — CHANGES_REQUESTED** [review]
The pagination and link stopped working :/

Screenshot 2023-02-27 at 18 20 35

File: apps/meteor/client/views/omnichannel/triggers/TriggersTableContainer.tsx [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+import TriggersTable from './TriggersTable';
+
+const TriggersTableContainer = () => {
+	const t = useTranslation();
+	const [params, setParams] = useState<GenericTableParams>(() => ({ current: 0, itemsPerPage: 25 }));
+
+	const { current, itemsPerPage } = params;
+
+	const getTriggers = useEndpoint('GET', '/v1/livechat/triggers');
+	const { data, refetch, isError } = useQuery(['/v1/livechat/triggers'], () => getTriggers({ offset: current, count: itemsPerPage }));
```
Comment (filipemarins) [inline apps/meteor/client/views/omnichannel/triggers/TriggersTableContainer.tsx:16]:
```suggestion
	const { data, refetch, isError } = useQuery(['/v1/livechat/triggers', current, itemsPerPage], () =>
		getTriggers({ offset: current, count: itemsPerPage }),
	);
```

I believe that you will need to pass the `current` and `itemPerPage` here as a key.

File: apps/meteor/client/views/omnichannel/triggers/TriggersTableContainer.tsx [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+
+	if (isError) {
+		return <Callout>{t('Error')}: error</Callout>;
+	}
+
+	return (
+		<TriggersTable
+			triggers={data?.triggers || []}
+			totalTriggers={data?.triggers.length || 0}
```
Comment (filipemarins) [inline apps/meteor/client/views/omnichannel/triggers/TriggersTableContainer.tsx:25]:
```suggestion
			totalTriggers={data?.total || 0}
```
This one has the correct total items. Also, you will need to update the `'/v1/livechat/triggers'` endpoint type return to add the `total: number` there.

**Review by filipemarins — APPROVED** [review]

**Review by sampaiodiego — CHANGES_REQUESTED** [review]
Are you guys really confident all these file migrations are just "regressions"?

I mean, rewriting a JS file into TS will never a fix for a regression.. other than that, we've seen many bugs being introduced by such rewrites.

so should a real regression be done instead? and let the file conversion to a separate PR?

**Comment by dougfabris** [comment]
Hi there, thanks for the contribution! 🚀 💯 

Closing this PR because it's stale now and it was already fixed in another one.

---

**Questions? Help needed? Feature Requests?** 

- Join our [Open Server](https://open.rocket.chat/) in the #support channel and feel free to raise a question
- Join our [Community Forum](https://forums.rocket.chat/) and search/create a post there

**Pull request closed without merging**